### ISSUE TITLE
Variable declaration with ! postfix operator results in invalid JS

### ISSUE THREAD

**Issue by aleclarson** [alangpierce/sucrase#406]
```ts
function example1() {
    let resolve: (value: any) => void
    let promise: any = new Promise(_resolve => {
        resolve = _resolve
    })
    // With "strictNullChecks", this is an error:
    promise.resolve = resolve
}

function example2() {
    // Unless we use an exclamation in the `let` declaration:
    let resolve!: (value: any) => void
    let promise: any = new Promise(_resolve => {
        resolve = _resolve
    })
    promise.resolve = resolve
}
```

- [Sucrase output][1]
- [Playground][2]

[1]: https://sucrase.io/#compressedCode=H4sIADTCP1wAA71QPQvCMBDd%2ByuenVoQi46FuriLizhqiFcaTJOSj6qI%2F92YagUHR4%2BDg3v33t272ivuhFagC2s7SfMsxy1BCEkOhqyWPZXIeiZ9qExdc1RL9Focx6nO6FbYAUUFRWdshla2fyk8OYPsM8Ym3niE7nksRYGdcA1S64zgbu2lXDXETzadwjXCIiQLBxujTRkZrwNmH9237D1J6i%2BHi9FhWLRVkqzFmeAtRdULl6xlkSBU2Ec4BI8HHCkAJgLl938m%2F33QL7sPsCfcwtABAAA%3D

[2]: http://www.typescriptlang.org/play/#src=function%20example1()%20%7B%0D%0A%20%20%20%20let%20resolve%3A%20(value%3A%20any)%20%3D%3E%20void%0D%0A%20%20%20%20let%20promise%3A%20any%20%3D%20new%20Promise(_resolve%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%20%20resolve%20%3D%20_resolve%0D%0A%20%20%20%20%7D)%0D%0A%20%20%20%20%2F%2F%20With%20%22strictNullChecks%22%2C%20this%20is%20an%20error%3A%0D%0A%20%20%20%20promise.resolve%20%3D%20resolve%0D%0A%7D%0D%0A%0D%0Afunction%20example2()%20%7B%0D%0A%20%20%20%20%2F%2F%20Unless%20we%20use%20an%20exclamation%20in%20the%20%60let%60%20declaration%3A%0D%0A%20%20%20%20let%20resolve!%3A%20(value%3A%20any)%20%3D%3E%20void%0D%0A%20%20%20%20let%20promise%3A%20any%20%3D%20new%20Promise(_resolve%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%20%20resolve%20%3D%20_resolve%0D%0A%20%20%20%20%7D)%0D%0A%20%20%20%20promise.resolve%20%3D%20resolve%0D%0A%7D%0D%0A

**Comment by alangpierce** [comment]
Thanks for reporting! I actually wasn't aware of that syntax, only the class variant, but makes sense. I guess I overlooked it from https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html

**Comment by alangpierce** [comment]
@aleclarson sorry for the delay, I just published as 3.9.6.
